app/uk/gov/hmrc/essttpstubs/services/AffordableQuotesService.scala

```diff
@@ -0,0 +1,164 @@
+    val optimumPlan: PaymentPlan = optimumPaymentPlan(affordableQuotesRequest.paymentPlanAffordableAmount, allPossiblePlans) match {
+      case Left(_)      => allPossiblePlans.find(p => p.planDuration.value === 1).getOrElse(throw new RuntimeException("There was no optimum plan, investigate..."))
+      case Right(value) => value
+    }
+    val paymentPlans = deriveCollectionsToReturnBasedOnOptimum(optimumPlan, allPossiblePlans)
+    paymentPlans match {
+      case Right(valid)    => AffordableQuotesResponse(valid)
+      case Left(someError) => throw new RuntimeException(someError.message)
+    }
+  }
+
+  private def computeAllPossiblePlans(affordableQuotesRequest: AffordableQuotesRequest, totalDebt: TotalDebt, interestPerMonth: AmountInPence, initialCollection: Option[InitialCollection]): List[PaymentPlan] =
+    (1 to 6).map { numberOfInstalments =>
+      computeAPlan(numberOfInstalments, totalDebt, interestPerMonth, affordableQuotesRequest.paymentPlanStartDate, initialCollection, affordableQuotesRequest)
+    }.toList
+
+  private def computeAPlan(planDuration: Int, totalDebt: TotalDebt, interestPerMonth: AmountInPence, startDate: InstalmentStartDate, initialCollection: Option[InitialCollection], affordableQuotesRequest: AffordableQuotesRequest): PaymentPlan = {
+    val totalInterest = AmountInPence(interestPerMonth.value * planDuration)
+    val totalDebtPlusInterest = TotalDebtIncludingInterest(totalDebt.value.+(totalInterest))
+    val collections = computeCollections(planDuration, totalDebtPlusInterest, startDate, initialCollection)
+    PaymentPlan(
+      numberOfInstalments = NumberOfInstalments(planDuration),
+      planDuration        = PlanDuration(planDuration),
+      totalDebt           = totalDebt,
+      totalDebtIncInt     = totalDebtPlusInterest,
+      planInterest        = PlanInterest(totalInterest),
+      collections         = collections,
+      instalments         = computeInstalments(affordableQuotesRequest, totalDebt, interestPerMonth, collections.regularCollections.size)
+    )
+  }
+
+  private def computeCollections(planDuration: Int, totalDebtIncludingInterest: TotalDebtIncludingInterest, planStartDate: InstalmentStartDate, initialCollection: Option[InitialCollection]): Collection = {
+    val amountDue = AmountDue(AmountInPence(totalDebtIncludingInterest.value.value / planDuration))
+    Collection(
+      initialCollection  = initialCollection,
+      regularCollections = (1 to planDuration).map { x =>
+        RegularCollection(
+          DueDate(planStartDate.value.plusMonths(x)),
+          amountDue
+        )
+      }.toList
+    )
+  }
+
+  //optimum is the one with closest monthly amount from collection, which is less than or equal to affordable amount
+  def optimumPaymentPlan(paymentPlanAffordableAmount: PaymentPlanAffordableAmount, paymentPlans: List[PaymentPlan]): Either[AffordableQuotesService.CalculationError, PaymentPlan] = {
```

Comment:
I think you can probably do away with `CalculationError` and just use an `Option` here. `CalculationError` sounds like the whole calculation is bust but there's still something being done in the non-happy case here afterwards so it's not a total disaster.